### PR TITLE
drake-factual: init at 1.0.3

### PR DESCRIPTION
**Pull request by rybern**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by rybern** [comment]
This one was pretty annoying. I couldn't find any other examples of packages that build with leiningen. Leiningen installs dependencies in a local Maven cache, which is by default /var/empty/.m2. This can be changed with a variable in the project.clj file. That's what the sed command does.

File: pkgs/development/tools/build-managers/drake-factual/default.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    mv bin/drake $out/bin
+    mv target/drake-${numVersion}.jar $out/target
+  '';
+
+  phases = [ "unpackPhase" "patchPhase" "configurePhase" "buildPhase" "checkPhase"];
+
+  meta = with stdenv; {
+    description = "a simple-to-use, extensible, text-based data workflow tool that organizes command execution around data and its dependencies";
+    homepage = https://github.com/Factual/drake;
+    license = licenses.epl;
```
Comment (pbogdan) [inline pkgs/development/tools/build-managers/drake-factual/default.nix]: `lib.licenses.epl10`

File: pkgs/development/tools/build-managers/drake-factual/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    platforms = lib.platforms.linux;
+    maintainers = with maintainers; [ rybern ];
```
Comment (pbogdan) [inline pkgs/development/tools/build-managers/drake-factual/default.nix]: `lib.maintainers`

**Comment by grahamc** [comment]
@grahamcofborg eval

File: pkgs/development/tools/build-managers/drake-factual/default.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+  src = fetchFromGitHub {
+    repo = pname;
+    owner = "Factual";
+    rev = version;
+    sha256 = "0sm8ghw9x7ba05q6c2gx1my46xigw0a0djjdlrqni2jj3pf0v27w";
+  };
+
+  buildInputs = [ leiningen ];
+  propagatedBuildInputs = [ jdk which ];
```
Comment (Mic92) [inline pkgs/development/tools/build-managers/drake-factual/default.nix]: `propagatedBuildInputs` is not the right way of specifying run-time dependencies. Use makeWrapper as described in https://nixos.org/nixpkgs/manual/#sec-language-java

**Comment by yegortimoshenko** [comment]
Also, this won't build in Nix sandbox. You will need a fixed-output derivation, you can check one of the versions of `fetchMaven` (should be renamed to `fetchLeiningen`) that handles Leiningen dependency fetches:

https://github.com/yegortimoshenko/overlay/blob/4bf36bc2bf6f8756eae27e1d3afd31e6aac9dbf5/pkgs/fetchMaven/default.nix

**Review by matthewbauer — CHANGES_REQUESTED** [review]
See @mic92's comment

**Comment by jtojnar** [comment]
@rybern Are you still interested in this?

**Pull request closed without merging**